### PR TITLE
Fix issue 2104 - escape function for regular expressions

### PR DESCRIPTION
**Pull request by DmitryOlshansky**
The oldest enhancement request for std.regex...

### CODE REVIEW

File: std/regex/package.d [hunk 1]
```diff
@@ -1426,3 +1426,61 @@ public @trusted String[] split(String, RegEx)(String input, RegEx rx)
 
 ///Exception object thrown in case of errors during regex compilation.
 public alias RegexException = std.regex.internal.ir.RegexException;
+
+/++
+  A range that lazily produces an string output escaped to be used inside of a regular expression.
++/
+auto escaper(Range)(Range r)
+{
+    import std.algorithm : find;
```
Comment (9il) [inline std/regex/package.d]: more detailed import please

File: std/regex/package.d [hunk 2]
```diff
@@ -1426,3 +1426,61 @@ public @trusted String[] split(String, RegEx)(String input, RegEx rx)
...
+    static immutable escapables = [Escapables];
+    static struct Escaper(Range) // template to deduce attributes
```
Comment (9il) [inline std/regex/package.d]: but methods are already templates

File: std/regex/package.d [hunk 3]
```diff
@@ -1426,3 +1426,61 @@ public @trusted String[] split(String, RegEx)(String input, RegEx rx)
 
 ///Exception object thrown in case of errors during regex compilation.
 public alias RegexException = std.regex.internal.ir.RegexException;
+
+/++
+  A range that lazily produces an string output escaped to be used inside of a regular expression.
```
Comment (schuetzm) [inline std/regex/package.d]: "a string output"

**Comment by DmitryOlshansky** [comment]
Addressed all comments

**Comment by 9il** [comment]
fails

**Comment by DmitryOlshansky** [comment]
> fails

Fixed

**Comment by 9il** [comment]
Auto-merge toggled on

**Pull request merged**